File: apps/mobile/ios/Uniswap/Notifications/SilentPushEventEmitter.m
//
//  SilentPushEventEmitter.m
//  Uniswap
//
//  Created by John Short on 9/29/25.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(SilentPushEventEmitter, RCTEventEmitter)

RCT_EXTERN_METHOD(supportedEvents)
RCT_EXTERN_METHOD(addListener:(NSString *)eventName)
RCT_EXTERN_METHOD(removeListeners:(nonnull NSNumber *)count)

@end

File: apps/mobile/ios/Uniswap/Notifications/SilentPushEventEmitter.swift
//
//  SilentPushEventEmitter.swift
//  Uniswap
//
//  Created by John Short on 9/29/25.
//

import React

@objc(SilentPushEventEmitter)
open class SilentPushEventEmitter: RCTEventEmitter {

  public static weak var emitter: RCTEventEmitter?

  override init() {
    super.init()
    SilentPushEventEmitter.emitter = self
  }

  open override func supportedEvents() -> [String] {
    ["SilentPushReceived"]
  }

  @objc(emitEventWithPayload:)
  public static func emitEvent(with payload: [String: Any]) {
    guard let emitter = emitter else {
      return
    }
    emitter.sendEvent(withName: "SilentPushReceived", body: payload)
  }
}
